File: laravel/app/Models/Camera.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Camera extends Model
{
    use HasFactory;

    protected $fillable = [
        'store_id',
        'name',
        'ip_address',
        'port',
        'username',
        'password',
        'server_port',
        'is_online',
        'last_online_at',
        'last_downloaded_at',
        'last_error',
        'last_error_at',
        'last_health_check_at',
    ];

    protected $casts = [
        'is_online' => 'boolean',
        'port' => 'integer',
        'server_port' => 'integer',
        'last_online_at' => 'datetime',
        'last_downloaded_at' => 'datetime',
        'last_error_at' => 'datetime',
        'last_health_check_at' => 'datetime',
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
    ];

    protected $hidden = [
        'password',
    ];

    public function store(): BelongsTo
    {
        return $this->belongsTo(Store::class);
    }

    public function fileDownloadJobs(): HasMany
    {
        return $this->hasMany(FileDownloadJob::class);
    }

    public function getStatusAttribute(): string
    {
        if ($this->fileDownloadJobs()->where('status', 'downloading')->exists()) {
            return 'Downloading';
        }
        return 'Idle';
    }
}

File: HikvisionService/Models/Camera.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HikvisionService.Models;

[Table("cameras")]
public class Camera
{
    [Key]
    [Column("id")]
    public long Id { get; set; }

    [Column("store_id")]
    public long? StoreId { get; set; }

    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [Column("ip_address")]
    public string IpAddress { get; set; } = string.Empty;

    [Column("port")]
    public int Port { get; set; } = 554;

    [Column("username")]
    public string? Username { get; set; }

    [Column("password")]
    public string? Password { get; set; }

    [Column("server_port")]
    public int? ServerPort { get; set; }

    [Column("is_online")]
    public bool IsOnline { get; set; } = false;

    [Column("last_online_at")]
    public DateTime? LastOnlineAt { get; set; }

    [Column("last_downloaded_at")]
    public DateTime? LastDownloadedAt { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; }

    // Navigation property
    public virtual Store? Store { get; set; }
    public virtual ICollection<FileDownloadJob> FileDownloadJobs { get; set; } = new List<FileDownloadJob>();
}